gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerRequestManager.java

```diff
@@ -115,37 +116,57 @@ private static boolean responseValidation(final DirectBuffer responseContent) {
       nodeIdProvider = determineBrokerNodeIdProvider(request);
     } catch (final PartitionNotFoundException e) {
       returnFuture.completeExceptionally(e);
+      GatewayMetrics.registerFailedRequest(request.getPartitionId(), request.getType());
       return;
     }
 
     final ActorFuture<DirectBuffer> responseFuture =
         sender.send(clientTransport, nodeIdProvider, request, requestTimeout);
+    final long startTime = System.currentTimeMillis();
 
     actor.runOnCompletion(
         responseFuture,
         (clientResponse, error) -> {
           try {
             if (error == null) {
               final BrokerResponse<T> response = request.getResponse(clientResponse);
-              handleResponse(response, returnFuture);
+
+              if (handleResponse(response, returnFuture)) {
```

Comment:
Considering "RESOURCE_EXHAUSTED" as successful responses can skew the elapsedtime metrics as it is usually much smaller than a successfully processed requests.